File: Library/src/main/java/com/sunzn/cnki/library/Source.java
package com.sunzn.cnki.library;

public enum Source {

    JOU("J", "期刊"), DOC("D", "博士"), MAS("M", "硕士"), CON("C", "会议"), NEW("N", "报纸"), UNK("U", "未知");

    private String code;
    private String name;

    Source(String code, String name) {
        this.code = code;
        this.name = name;
    }

    public String getCode() {
        return code;
    }


    public String getName() {
        return name;
    }

    public static Source route(String value) {
        if (value != null && value.length() > 0) {
            if (value.contains(JOU.getName())) {
                return JOU;
            }
            if (value.contains(DOC.getName())) {
                return DOC;
            }
            if (value.contains(MAS.getName())) {
                return MAS;
            }
            if (value.contains(CON.getName())) {
                return CON;
            }
            if (value.contains(NEW.getName())) {
                return NEW;
            }
        }
        return UNK;
    }

}

File: Library/src/main/java/com/sunzn/cnki/library/Source.kt
package com.sunzn.cnki.library

enum class Source(val code: String, val source: String) {

    JOU("J", "期刊"),
    DOC("D", "博士"),
    MAS("M", "硕士"),
    CON("C", "会议"),
    NEW("N", "报纸"),
    UNK("U", "未知");

    companion object {
        fun route(value: String?): Source {
            if (value != null && value.isNotEmpty()) {
                if (value.contains(JOU.source)) {
                    return JOU
                }
                if (value.contains(DOC.source)) {
                    return DOC
                }
                if (value.contains(MAS.source)) {
                    return MAS
                }
                if (value.contains(CON.source)) {
                    return CON
                }
                if (value.contains(NEW.source)) {
                    return NEW
                }
            }
            return UNK
        }
    }

}
